kotsadm/pkg/apiserver/server.go

```diff
@@ -102,6 +102,7 @@ func Start() {
 	// Apps
 	r.Path("/api/v1/apps").Methods("OPTIONS", "GET").HandlerFunc(handlers.ListApps)
 	r.Path("/api/v1/apps/app/{appSlug}").Methods("OPTIONS", "GET").HandlerFunc(handlers.GetApp)
+	r.Path("/api/v1/app/{appSlug}/cluster/{clusterSlug}").Methods("OPTIONS", "GET").HandlerFunc(handlers.GetAppVersionHistory) // NOTE: why do we need cluster slug here?
```

Comment:
how can i avoid this? clusterSlug

Reply:
1- in the api: list the downstreams for the app and get the first one. remove clusterSlug from the path
2- in the UI: don't send it?